Adds CLI for Microsoft 365 v10.11 release blog post (#2148)
pnp/blog: content/cli-for-microsoft-365/cli-for-microsoft-365-v10-11/index.md

---
title: CLI for Microsoft 365 v10.11
date: 2025-08-31T01:00:00.000Z
author: Adam Wójcik
githubname: adam-it
categories:
  - CLI for Microsoft 365
images:
  - images/banner-cli-m365.png
tags:
  - CLI for Microsoft 365
  - SharePoint
  - SharePoint Embedded
  - Microsoft Teams
  - Viva Engage
type: popular
---

The CLI for Microsoft 365 is back with a fresh minor release! We’ve just launched a new version of [CLI for Microsoft 365](https://aka.ms/cli-m365), a versatile cross-platform command-line tool for managing your Microsoft 365 tenant and SharePoint Framework projects.

> Explore the [release notes](https://aka.ms/cli-m365/notes) for a neat overview of all the exciting features and improvements. 

## What's New

### SharePoint Embedded

We are continuously building and improving SharePoint Embedded management commands in CLI for Microsoft 365. This time, we managed to add a new command that will allow you to restore a deleted container.

You may restore a container by ID:

```sh
m365 spe container recyclebinitem restore --id "b!ISJs1WRro0y0EWgkUYcktDa0mE8zSlFEqFzqRn70Zwp1CEtDEBZgQICPkRbil_5Z"
```

> Or using something more 'human' friendly, using its name and container type name.

Reviewer: ```suggestion
Or using something more 'human'-friendly, using its name and container type name.
```

```sh
m365 spe container recyclebinitem restore --containerTypeName "My container type name" --name "Invoices"
```

For more information, check the following resource:

- [spe container recyclebinitem restore](https://pnp.github.io/cli-microsoft365/cmd/spe/container/container-recyclebinitem-restore/)

### Viva Engage

> This release also brings two new commands to Viva Engage area. Now you may list all Viva Engage roles as well as list all users assigned to a specific role.

Reviewer: ```suggestion
This release also brings two new commands to the Viva Engage area. Now you may list all Viva Engage roles as well as list all users assigned to a specific role.
```

To list all Viva Engage roles, run:

```sh
m365 viva engage role list
```

And to get all users assigned to a Viva Engage role specified by name, run the following:

```sh
m365 viva engage role member list --roleName 'Verified Admin'
```

> For more information, check the following resource:

Reviewer: ```suggestion
For more information, check the following resources:
```

- [viva engage role member list](https://pnp.github.io/cli-microsoft365/cmd/viva/engage/engage-role-list/)
- [viva engage role list](https://pnp.github.io/cli-microsoft365/cmd/viva/engage/engage-role-list/)

> ### Microsoft Teams    

Reviewer: ```suggestion
### Microsoft Teams
```

> This release brings a fresh new command allowing you to list all Teams calls within your tenant. Retrieving call records was never easier.

Reviewer: ```suggestion
This release brings a fresh new command allowing you to list all Teams calls within your tenant. Retrieving call records has never been easier.
```

To get all call records from the past 30 days, simply run:

```sh

> m365 teams call record list

Reviewer 2: ```suggestion
m365 teams callrecord list
```

```

Or to get all call records where a specific user participated, run:

```sh
m365 teams callrecord list --userName "john.doe@contoso.com"
```

It is also possible to get all call records within a specific timeframe.

```sh

> m365 teams callrecord list --startDateTime "2025-05-1T00:00:00Z" --endDateTime "2025-05-14T00:00:00Z"

Reviewer 2: ```suggestion
m365 teams callrecord list --startDateTime "2025-05-01T00:00:00Z" --endDateTime "2025-05-14T00:00:00Z"
```

To create a new app registration to use for CLI for Microsoft 365, run:

```sh
m365 cli app add
```

In order to create a new app registration with all permission scopes and a custom name, run the following:

```sh
m365 cli app add --name "Contoso CLI app" --scopes all
```

Lastly, if you want to create a new app registration with self-defined scopes, just execute:

```sh
m365 cli app add --scopes "https://graph.microsoft.com/User.Read,https://graph.microsoft.com/Group.Read.All"
```

For more information, check the following resource:

- [cli app add](https://pnp.github.io/cli-microsoft365/cmd/cli/app/app-add/)

## What's changed

> This release also brings some other great changes. Aside from many improvements in the codebase and bug fixes to elevate the overall CLI experience, there are several important updates we'd like to highlight.

Reviewer 2: Another section we could add is about support for SPFx v1.22.0-beta.1. I’m not entirely sure though if we typically include beta support updates for SPFx in the release notes. 😄

> ### Add subgroups using entra group member add command

Reviewer: I'd make it a bit more generic since the command in the title makes it harder to read.

```suggestion
### Nesting Entra ID groups
```

In this release, it's also worth mentioning that we updated the [entra group member add](https://pnp.github.io/cli-microsoft365/cmd/entra/group/group-member-add/), allowing you to add subgroups as members of an Entra group. 

> For example, you may now add multiple subgroups specified by name as members to a group specified by ID, by running:

Reviewer: ```suggestion
For example, you may now add multiple subgroups specified by name as members of a group specified by ID, by running:
```

```

For more information, check the following resource:

- [spe container recyclebinitem restore](https://pnp.github.io/cli-microsoft365/cmd/spe/container/container-recyclebinitem-restore/)

### Viva Engage

This release also brings two new commands to the Viva Engage area. Now you may list all Viva Engage roles as well as list all users assigned to a specific role.

To list all Viva Engage roles, run:

```sh
m365 viva engage role list
```

And to get all users assigned to a Viva Engage role specified by name, run the following:

```sh
m365 viva engage role member list --roleName 'Verified Admin'
```

For more information, check the following resources:

- [viva engage role member list](https://pnp.github.io/cli-microsoft365/cmd/viva/engage/engage-role-list/)
- [viva engage role list](https://pnp.github.io/cli-microsoft365/cmd/viva/engage/engage-role-list/)

### Microsoft Teams

This release brings a fresh new command allowing you to list all Teams calls within your tenant. Retrieving call records has never been easier.

To get all call records from the past 30 days, simply run:

```sh

> m365 teams call record list

Author: ```suggestion
m365 teams callrecord list
```
